src/harness/unittests/organizeImports.ts

```diff
@@ -0,0 +1,431 @@
+            const imports = filter(sourceFile.statements, isImportDeclaration);
+            assert.equal(importStrings.length, imports.length);
+            return imports;
+        }
+
+        function assertEqual(node1?: Node, node2?: Node) {
+            if (node1 === undefined) {
+                assert.isUndefined(node2);
+                return;
+            }
+            else if (node2 === undefined) {
+                assert.isUndefined(node1); // Guaranteed to fail
+                return;
+            }
+
+            assert.equal(node1.kind, node2.kind);
+
+            switch (node1.kind) {
+                case SyntaxKind.ImportDeclaration:
+                    const decl1 = node1 as ImportDeclaration;
+                    const decl2 = node2 as ImportDeclaration;
+                    assertEqual(decl1.importClause, decl2.importClause);
+                    assertEqual(decl1.moduleSpecifier, decl2.moduleSpecifier);
+                    break;
+                case SyntaxKind.ImportClause:
+                    const clause1 = node1 as ImportClause;
+                    const clause2 = node2 as ImportClause;
+                    assertEqual(clause1.name, clause2.name);
+                    assertEqual(clause1.namedBindings, clause2.namedBindings);
+                    break;
+                case SyntaxKind.NamespaceImport:
+                    const nsi1 = node1 as NamespaceImport;
+                    const nsi2 = node2 as NamespaceImport;
+                    assertEqual(nsi1.name, nsi2.name);
+                    break;
+                case SyntaxKind.NamedImports:
+                    const ni1 = node1 as NamedImports;
+                    const ni2 = node2 as NamedImports;
+                    assertListEqual(ni1.elements, ni2.elements);
+                    break;
+                case SyntaxKind.ImportSpecifier:
+                    const is1 = node1 as ImportSpecifier;
+                    const is2 = node2 as ImportSpecifier;
+                    assertEqual(is1.name, is2.name);
+                    assertEqual(is1.propertyName, is2.propertyName);
+                    break;
+                case SyntaxKind.Identifier:
+                    const id1 = node1 as Identifier;
+                    const id2 = node2 as Identifier;
+                    assert.equal(id1.text, id2.text);
+                    break;
+                case SyntaxKind.StringLiteral:
+                case SyntaxKind.NoSubstitutionTemplateLiteral:
+                    const sl1 = node1 as LiteralLikeNode;
+                    const sl2 = node2 as LiteralLikeNode;
+                    assert.equal(sl1.text, sl2.text);
+                    break;
+                default:
+                    assert.equal(node1.getText(), node2.getText());
+                    break;
+            }
+        }
+
+        function assertListEqual(list1: ReadonlyArray<Node>, list2: ReadonlyArray<Node>) {

```

Comment:
I think it would be simpler to emit and compare strings, vs parsing and comparing ASTs. That way there's no need to write `assertEqual`  and we also test if we are correctly formatting the result. Also would make error messages more readable.

Reply:
Suits me.  How do I stringify them?

Reply:
I'm attempting to do this, but the emitter works best if the ast it's working on is either *entirely* synthesized, or not at all. I'm debugging and it looks like there are some nodes with `pos` set  as children of a synthesized node, which causes the emitter to look at the source file for comments, only there is no source text. Some part of organizeImports might be forgetting to clone a node?

Reply:
Regarding testing with text, it seems not to have been simpler.  I'm going to leave things as they are until we have a concrete problem to correct or improvement to make.

I was under the impression that the baseline tests were doing an end-to-end test.  Did you mean something else?

Reply:
No, baseline tests will do.